### ISSUE TITLE
Reporting which toolkit was used.

### ISSUE THREAD

**Issue by jthorton** [openforcefield/qcsubmit#1]
**Description**
In order to ensure that workflows are reproducible, we would like to report which back end toolkit and its version was used during the execution of a function.  See here for details on the current discussion on how this could be implemented https://github.com/openforcefield/openforcefield/issues/540.

**Outcome**
For now, while in early development we have chosen to use open eye explicitly for all back end toolkit calls until this is resolved.

**Comment by jchodera** [comment]
Could this be included in a metadata block in the emitted JSON?

Perhaps we could even have `qcsubmit` optionally _accept_ the same JSON block (or dict equivalent) in order to ensure the same backends are used if we want to make things reproducible.

**Comment by jthorton** [comment]
Yes the idea is to have each workflow component report what backend operations it used if any and include them in the metadata. Working it the other way around and accepting it would be possible if we give each component a toolkit argument.
